filesystem/android_super.ksy

```diff
@@ -0,0 +1,200 @@
+meta:
+  id: android_super
+  title: Android Dynamic Partitions metadata
+  application: Android
+  file-extension: img
+  license: CC0-1.0
+  ks-version: 0.9
+  bit-endian: le
+  endian: le
+
+doc: |
+  The metadata stored by Android at the beginning of a "super" partition, which
+  is what it calls a disk partition that holds one or more Dynamic Partitions.
+  Dynamic Partitions do more or less the same thing that LVM does on Linux,
+  allowing Android to map ranges of non-contiguous extents to a single logical
+  device. This metadata holds that mapping.
+
+doc-ref:
+  - https://source.android.com/devices/tech/ota/dynamic_partitions/implement
+  - https://android.googlesource.com/platform/system/core/+/master/fs_mgr/liblp/include/liblp/metadata_format.h
+
+seq:
+  - id: padding
+    size: 0x1000
+    type: dummy
+
+  - id: primary_geometry
+    size: 0x1000
+    type: geometry
+
+  - id: backup_geometry
+    type: geometry
+    size: 0x1000
+
+  - id: primary_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+  - id: backup_metadata
+    size: primary_geometry.metadata_max_size
+    type: metadata
+    repeat: expr
+    repeat-expr: primary_geometry.metadata_slot_count
+
+types:
+  dummy: {}
+
+  geometry:
+    seq:
+      - id: magic
+        contents: 'gDla'
+      - id: struct_size
+        type: u4
+      - id: checksum
+        size: 32
+      - id: metadata_max_size
+        type: u4
+      - id: metadata_slot_count
+        type: u4
+      - id: logical_block_size
+        type: u4
+
+  metadata:
+    seq:
+      - id: magic
+        contents: '0PLA'
+      - id: major_version
+        type: u2
+      - id: minor_version
+        type: u2
+      - id: header_size
+        type: u4
+      - id: header_checksum
+        size: 32
+        doc: |
+          SHA-256 hash of header_size bytes from beginning of metadata,
+          calculated as if header_checksum were zeroed out
+      - id: tables_size
+        type: u4
+      - id: tables_checksum
+        size: 32
+        doc: SHA-256 hash of tables_size bytes from end of header
+      - id: partitions
+        type: table_descriptor(table_kind::partitions)
+      - id: extents
+        type: table_descriptor(table_kind::extents)
+      - id: groups
+        type: table_descriptor(table_kind::groups)
+      - id: block_devices
+        type: table_descriptor(table_kind::block_devices)
+
+    enums:
+      table_kind:
+        0: partitions
+        1: extents
+        2: groups
+        3: block_devices
+
+    types:
+      table_descriptor:
+        params:
```

Comment:
```yaml
-affected-by: 135
```

Reply:
I don't see how you resolved this, @tchebb? The point is to provide a reference to an issue in the https://github.com/kaitai-io/kaitai_struct repository from the KSY that's blocking or impairing some feature of the particular KSY spec implementation. It's something like `TODO` or `FIXME when feature ... is available in KS language` comments in the programming world.

It's definitely a good idea, because it's true that https://github.com/kaitai-io/kaitai_struct/issues/135 would help here. I'm not sure though how it should be done: maybe a simple `doc` key would be better than this not-much-established `-affected-by` key _(I searched GitHub with term [`extension:ksy affected-by OR -affected-by`](https://github.com/search?q=extension%3Aksy+affected-by+OR+-affected-by) and found nothing, not sure, but e.g. [`extension:ksy -orig-id OR mac-os-resource-type`](https://github.com/search?q=extension%3Aksy+-orig-id+OR+mac-os-resource-type) works, so the former probably works as well)_. It's likely that readers of the spec wouldn't understand this key, and it's naturally not documented anywhere.

Instead, I would suggest something like this, which basically says `-affected-by: 188`:

https://github.com/kaitai-io/kaitai_struct_formats/blob/53795d043f170e2f2654b5a5774a48fadb1d349e/image/bmp.ksy#L104-L109

Reply:
>It's likely that readers of the spec wouldn't understand this key, and it's naturally not documented anywhere.

It is likely that it is currently only used in my specs, and my specs using it are not currently merged. Using `doc` for it is IMHO consumes too much space. The original purpose of `-affected-by` was for search and automation. I mean some bot that scans the repo on issues being closed and mentioning all the specs that should be fixed.

Reply:
> The original purpose of `-affected-by` was for search and automation. I mean some bot that scans the repo on issues being closed and mentioning all the specs that should be fixed.

That's a nice idea, thanks!

Reply:
IMHO since issues in kaitai-io/kaitai_struct are usually meant, for that repo it makes sense to use just an issue number.

Reply:
I'm not sure where you want me to put this line. Isn't the compiler going to complain about an unknown key if I add it right here?

Reply:
> I'm not sure where you want me to put this line. Isn't the compiler going to complain about an unknown key if I add it right here?

No, see https://doc.kaitai.io/user_guide.html#orig-id:

> Kaitai Struct compiler would just ignore any key that starts with `-`, and silently allow it. These kind of keys can be used to store arbitrary additional information, which can be accessible to external tools (i.e. other than the compiler). Feel free to add more arbitrary keys if you need to store extra structured information for some reason. (...)

Reply:
Ah, I see now: there's not supposed to be a space between the dash and `affected-by`. Added.